### PR TITLE
test: deflake TestBackoffRetryCount

### PR DESCRIPTION
**Pull request by Camila-B**
* This updates TestBackoffRetryCount so that it uses a timeout of 1 second instead of the reimport count when calling the `Run` function

### CODE REVIEW

File: pkg/parse/run_test.go [hunk 1]
```diff
@@ -427,41 +427,30 @@ func TestRun(t *testing.T) {
...
 	testState := &namespacecontroller.State{}
-	reimportCount := 0
 	retryCount := 0
-	testIsDone := func(reimportCount *int, _ *int) bool {
-		return *reimportCount == 35
-	}
 
 	t.Logf("start running test at %v", time.Now())
 
-	ctx, cancel := context.WithCancel(context.Background())
```
Comment (karlkfi) [inline pkg/parse/run_test.go:440]: defer the cancel to make sure the timer is cleaned up at the end of the function, even if it panics, without waiting for the timeout.

**Review by karlkfi — COMMENTED** [review]
This looks potentially less flaky, but still not deterministic. If the machine is busy it could start the timer and then pause the thread for more than a second before continuing. This is common on CI clusters with many jobs running in parallel. 

Is there a way we can refactor the test to test it in a way that's not time based? One option would be to inject named timer factories instead of named timeouts.

File: pkg/parse/run_test.go (hunk 1 above)
Comment (Camila-B) [reply pkg/parse/run_test.go:440]: Done

**Pull request closed without merging**